### PR TITLE
Modify S2078(multiple languages): Update to the education framework (APPSEC-115)

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**

### CODE REVIEW

File: rules/S2078/csharp/rule.adoc [hunk 1]
```diff
@@ -1,73 +1,36 @@
...
+== Comments And Links
+(visible only on this page)
+
+include::../comments-and-links.adoc[]
+endif::env-github,rspecator-view[]
+
+
+
 
 include::../see.adoc[]
```
Comment (loris-s-sonarsource) [inline rules/S2078/csharp/rule.adoc:35]:
```suggestion
```

File: rules/S2078/csharp/rule.adoc [hunk 2]
```diff
@@ -1,75 +1,19 @@
...
-            ds.Filter = "(&(uid=" + user + ")(userPassword=" + pass + "))"; // Noncompliant
+== Resources
 
-            // If the special value "*)(uid=*))(|(uid=*" is passed as user, authentication is bypassed
-            // Indeed, if it is passed as a user, the filter becomes:
-            // (&(uid=*)(uid=*))(|(uid=*)(userPassword=...))
-            // as uid=* match all users, it is equivalent to:
-            // (|(uid=*)(userPassword=...))
-            // again, as uid=* match all users, the filter becomes useless
+include::../common/resources/articles.adoc[]
```
Comment (loris-s-sonarsource) [inline rules/S2078/csharp/rule.adoc:13]:
```suggestion
```

File: rules/S2078/impact.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
+=== What is the possible impact?
+
+In the context of a web application vulnerable to LDAP injection: +
+After discovering the injection point, attackers insert data into the
```
Comment (pedro-oliveira-sonarsource) [inline rules/S2078/impact.adoc:4]:
```suggestion
In the context of a web application vulnerable to LDAP injection, after discovering the injection point, attackers insert data into the
```

File: rules/S2078/impact.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+==== Data leakage or corruption
+
+In typical scenarios where systems perform innocuous LDAP operations to find
+users or create inventories, an LDAP injection could result in data 
+leakage or corruption.
+
+==== Privilege escalation
+
+A malicious LDAP query could allow an attacker to impersonate an average user
+or administrator in scenarios where systems perform authorization checks or
```
Comment (pedro-oliveira-sonarsource) [inline rules/S2078/impact.adoc:22]:
```suggestion
A malicious LDAP query could allow an attacker to impersonate a low-privileged user
or an administrator in scenarios where systems perform authorization checks or
```

File: rules/S2078/common/fix/validation.adoc [hunk 5]
```diff
@@ -0,0 +1,45 @@
+==== Validation
+
+
+As a rule of thumb, the best approach to protect against injections is to
+systematically ensure that untrusted data cannot break out of the initially
+intended logic.
+
+For LDAP injections, the cleanest way to do so is to use parameterized queries
+if it is available for your use case. The second cleanest way to do so is to
```
Comment (pedro-oliveira-sonarsource) [inline rules/S2078/common/fix/validation.adoc:9]:
```suggestion
if it is available for your use case. An alternative is to
```

File: rules/S2078/csharp/how-to-fix-it/dotnet.adoc [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            return BadRequest();
+        }
+
+        DirectoryEntry directory  = new DirectoryEntry("LDAP://ou=system");
+        DirectorySearcher search  = new DirectorySearcher(directory);
+
+        search.Filter = "(&(uid=" + user + ")(userPassword=" + pass + "))";
+
+        return Json(ds.FindOne() != null);
```
Comment (pedro-oliveira-sonarsource) [inline rules/S2078/csharp/how-to-fix-it/dotnet.adoc:44]:
```suggestion
        return Json(search.FindOne() != null);
```

**Review by pedro-oliveira-sonarsource — APPROVED** [review]
LGTM. Small fix on dotnet.adoc and some minor final suggestions.

File: rules/S2078/impact.adoc (hunk 3 above)
Comment (loris-s-sonarsource) [reply rules/S2078/impact.adoc:4]: Ah, I'm not aligned with this suggestion, because it makes the paragraph bigger, and an even-bigger sentence

Comment (loris-s-sonarsource) [reply rules/S2078/impact.adoc:4]:
```suggestion
In the context of a web application vulnerable to LDAP injection:
after discovering the injection point, attackers insert data into the
```

Comment (loris-s-sonarsource) [reply rules/S2078/impact.adoc:4]: Agreed on this last syntactically correct fix

**Comment by loris-s-sonarsource** [comment]
All green ! I'm merging

**Pull request merged**